### DISCUSSION TITLE
Static rendering one part of the header while dynamically rendering the other half.

### DISCUSSION THREAD

**Discussion by akshityagi** [vercel/next.js#19787] [Help]
Hey,

My header has 2 parts. The left side has data that remains unchanged (Departments with products (Men, Women, Home) for my e-commerce website) and I want it to be statically rendered. While the right side has dynamic data (Cart and account info) and I want it to be dynamically rendered. Is there a way to achieve this?

Since the header is present on every page, there are a lot of query calls made on each render to our Database to render the left side. I want to avoid that.

Thanks,
Akshi Tyagi

<img width="1408" alt="Screenshot 2020-12-03 at 5 03 27 PM" src="https://user-images.githubusercontent.com/60875807/101013443-4ab3fa00-358a-11eb-9bd6-70c605832d9b.png">

**Comment by Tomburgs** [comment]
For the component on the left side you'd include [`getStaticProps`](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation) which would request the categories and such at build time.

Not sure what's the problem you're having with the component on the right side, if you're making API calls on render it should already be plenty dynamic.

**Reply by heekodev** [reply]
Thanks for the response @Tomburgs. 

My doubt was just regarding the left side. I'll include getStaticProps and see if that solves the issue.
